src/dreyfus_index.erl

```diff
@@ -157,9 +159,43 @@ handle_call(info, _From, State) -> % obsolete
     Reply = info_int(State#state.index_pid),
     {reply, Reply, State}.
 
+handle_cast({ddoc_updated, DDocResult}, State) ->
+    #state{dbname = DbName, index=#index{sig=CurrentSig}, waiting_list = WaitList} = State,
+    IndexDeleted = case DDocResult of
+        {not_found, deleted} ->
+            true;
+        {ok, DDoc} ->
+            Index = get_index_with_sig(DDoc, CurrentSig),
+            case Index of
+                [] ->
+                    true;
+                _ ->
+                    false
+            end
+        end,
+        case IndexDeleted of
+            true ->
+                case WaitList of
+                    [] ->
+                        self() ! {index_deleted},
+                        {noreply, State};
+                    _ ->
+                        erlang:send_after(?INDEX_SHUTDOWN_DELAY, self(), {index_deleted}),
+                        {noreply, State}
+                end;
+            false ->
+                {noreply, State}
+        end;
```

Comment:
since we always return {noreply, State}, move it to after the case statement so we say it once.